php/libraries/UserPermissions.class.inc

```diff
@@ -90,11 +90,7 @@ class UserPermissions
         // Fill the permissions array of this UserPermissions object with the
         // values extracted for this user from the database.
         if (is_array($results) and !empty($results)) {
-            foreach ($results AS $row) {
-                if (empty($row['userID'])) {
-                    $this->permissions[$row['code']] = false;
-                    continue;
-                }
```

Comment:
This also appears to be a regression

Reply:
iirc I removed this because it's redundant. `$row['userID']` can be 0 which would trigger the empty check. But in that case, Line 94 will always be false because `$this->userID` is taken from the ID of the `users` table which will never be 0. If line 94 is false, then the `$this->permissions` array will be correctly set to be false for that permission code.

Reply:
This is intended to unset any permissions which were unchecked on the request, it has nothing to do with userID 0 (which as far as I know should never exist)

Reply:
That's what I'm saying, I think line 94 and 95 (in the new code) are logically equivalent to what I deleted. These two lines alone will unset the permissions.

Reply:
Just to be clear I meant 94 and 95 on the right-hand side, not in the deleted code.